### ISSUE TITLE
Raptor

### ISSUE THREAD

**Issue by HellDragon-HK** [GreatEmerald/UT3Vehicles#12]
Issues and notes on the Raptor from in-game testing

UT2004 Draw  - 1.0?
UT3    Draw  - 1.3
UT3Port Draw - 1.0 / 1.17 visual test by in flight but a guess as with it clipping into the ground when sized up I can't measure it by standing beside it

Headlights don't cast light, nothing is hitting the wall but then this also true for the stock UT2004 Raptor on my game

  - [Priority]  
  - [Try 1.17 Drawscale]
  - [Wing collision missing on port, I can walk right through them to the body]  
  - [Wings don't change position on UT3 until a good distance off the ground, on port they move as soon as you get in and clip through the ground]  
  - [Secondary]  
  - [Collision for the back on UT3 is a little before the tail wings, on the port it's in between them]  
  - [Port has the stock UT2004 Raptor headlights in the middle of it's body]  
  - [Port is missing the front lights as well as the string of lights on along the wing's bottom (while sitting, while in flight I guess it's the side of the wing)]  
  - [Missing lights on the exhaust/engine section]  
  - [Engine flames are not large enough on port or something, the missing smoke makes it hard to tell but something feels off] 
  - [Sound effects are not identical, doesn't sound like the UT3 Raptor, sounds more like the stock UT2004 one]  
  - [Extra]  
  - [Front collision on UT3 is a bit concave with the direct front being right a the tips of the gun barrels, on the port its in the middle of them]  
  - [Front cont. move to the side a little on UT3 and you can stand on the middle of the of the gun barrel where the port's collision is, so it's nearly right]  
  - [On UT3 the Raptor moves backward at almost or exactly the same speed as hoverboard on UT3, I understand this was probably to make the Raptor less powerful so maybe could be an optional mutator or something as I'm sure most of us prefer being able to move backward at full speed since AVRiLs are not rare on UT2004 and you often have the entire team firing them at you]

**Comment by HellDragon-HK** [comment]
Raptor Impact with Manta
-------UT3 Results--------
--Front Hit   - Massive knockback and Manta is sent spinning like it's on ice, no damage taken, easily pushed like it's on ice
--Side Hit    - Massive knockback and Manta spun 180 degrees, no damage taken, easily pushed like it's on ice
--Back Hit    - Massive knockback and Manta is sent spinning like it's on ice, no damage taken, easily pushed like it's on ice
--Diagnol Hit - Massive knockback and Manta is sent spinning like it's on ice, no damage taken, easily pushed like it's on ice
------UT2004 Results------ 
--Front Hit   - Massive knockback and Manta is sent spinning like it's on ice, a lot of damage taken by both vehicles, easily pushed like it's on ice
--Side Hit    - Massive knockback like it's on ice but didn't spin, a lot of damage taken by both vehicles, easily pushed like it's on ice
--Back Hit    - Extreme knockback and sent spinning like it's on ice, Manta took massive damage and Raptor also took a lot, easily pushed like it's on ice
--Diagnol Hit - 1st attempt (still using the same 2 vehicles since I started) both vehicles are destroyed as soon as Manta is tapped, 2nd attempt is massive knockback and Manta is sent spinning across the ground like it's on ice, Manta took a lot of damage and Raptor took some, easily pushed like it's on ice
Extra Note: In UT3 the Manta had to have taken some damage after all of that but I can't tell with PS3, Raptor didn't take any that I could see, clearly wasn't the case with UT2004

Raptor Impact with Scorpion
-------UT3 Results--------
--Front Hit   - Little knocback and Scorpion went vertical and flipped upside down while Raptor is deflected and control lost, no damage taken, cannot be pushed
--Side Hit    - Little knockback and Scorpion was rolled over to an upside down position, no damage taken, can be easily pushed while upside down only
--Back Hit    - Fair knockback and Scorpion is lifted off back wheels, no damage taken, cannot be pushed
--Diagnol Hit - Little knockback and Scorpion was rolled over to an upside down position and Raptor was deflected with slight control loss, no damage taken, can be pushed while upside down (usually it flips itself back over but not after a Raptor does it)
------UT2004 Results------ 
--Front Hit   - Fair knockback and Scorpion was lifted off front wheels, damage taken, can be pushed
--Side Hit    - Fair knockback and Scorpion almost flipped, damage taken, flips at the slightest tap and then can be pushed while on it's side
--Back Hit    - Fair knockback and Scorpion lifted off back wheels, damage taken, can be pushed
--Diagnol Hit - Fair knockback and Scorpion nearly flipped and turned 90 degrees, damage taken, cannot be pushed without flipping
Extra Note: In UT2004 Scorpion did not take damage but Raptor took a lot to the point I needed to repair it (going to start trying to do that from now on), in UT3 neither vehicle appeared to take damage and when they do it's very little, I'd say UT3 vehicles take a lot less damage than UT2004 vehicles from all damage sources

Raptor Impact with Hellbender
-------UT3 Results--------
--Front Hit   - Little knockback, no damage taken, cannot be pushed
--Side Hit    - Little knockback and Raptor is deflected, no damage taken, cannot be pushed
--Back Hit    - Little knockback and Hellbender is lifted off back wheels and turns slightly, no damage taken, cannot be pushed
--Diagnol Hit - Little knockback and Hellbender is lifted off 3 of it's wheels and spun 180 degrees and the Raptor is deflected with control loss, cannot be pushed but can almost and probably be flipped
------UT2004 Results------ 
--Front Hit   - Medium knockback and lifted off front wheels, damage taken, can be pushed
--Side Hit    - Fair knockback and nearly flipped and Raptor went bottom up, damage taken, can be nearly flipped and then pushed easily and it nearly looks like you could use the turret barrels to carry it off like a forklift
--Back Hit    - Fair knockback and lifted off back wheels and spun almost 180 degrees, damage taken, can be pushed
--Diagnol Hit - Massive knockback and lifted off front wheels spun around lifted off all wheels landed on front wheels bottom up and slid across the ground for a bit before coming back down and stopping (a spectaular sight), damage taken, can be pushed flipped and turned

Raptor Impact with Stealthbender
-------UT3 Results--------
--Front Hit   - Fair to Medium knockback and Stealthbender was lifted off 3 wheels and spun more than 90 degrees, no damage taken, cannot be pushed
--Side Hit    - Little knockback and was rolled over, no damage taken, cannot be pushed but can be flipped
--Back Hit    - Little knockback, no damage taken, cannot be pushed but can be flipped bottom up
--Diagnol Hit - No knockback and slighty turned and Raptor was deflected with slight control loss, no damage taken, cannot be pushed but can with difficulty be flipped on it's side
------UT2004 Results------ 
--Front Hit   - pending stealthbender stability fixes
--Side Hit    - pending stealthbender stability fixes
--Back Hit    - pending stealthbender stability fixes
--Diagnol Hit - pending stealthbender stability fixes

Raptor Impact with Raptor
-------UT3 Results--------
--Front Hit   - Little knockback and other Raptor turned 90 degrees while mine was deflected with control loss, damage taken, cannot be pushed
--Side Hit    - Fair knockback and other Raptor is lifted off the ground, damage taken, can be turned but not really pushed
--Back Hit    - Fair knockback, damage taken, can be pushed
--Diagnol Hit - about the same as a side hit
------UT2004 Results------ 
--Front Hit   - Extreme knockback, damage taken, easily pushed like it's on ice
--Side Hit    - Extreme knockback and sent spinning like it's on ice, damage taken, easily pushed like it's on ice
--Back Hit    - Very extreme knockback, damage taken, easily pushed like it's on ice
--Diagnol Hit - Medium knockback and sent spinning like it's on ice, damage taken, easily pushed like it's on ice

Raptor Impact with Cicada
-------UT3 Results--------
--Front Hit   - Little knockback but Cicada front is lifted off ground and it turns 90 degrees, no damage taken, cannot be pushed but can be lifted off the ground some
--Side Hit    - Medium knockback and Cicada is thrown into the air, damage taken, cannot be pushed
--Back Hit    - Fair knockback and Cicada is thrown into the air, no damage taken, cannot be pushed but can be turned
--Diagnol Hit - Fair knockback and Cicada is thrown into the air and turns 90 degrees, no damage taken, can be pushed with difficulty but easily turned
------UT2004 Results------ 
--Front Hit   - Fair knockback and some spin, damage taken, easily pushed
--Side Hit    - Medium knockback and some almost 90 degreen spin, damage taken, easily pushed
--Back Hit    - Fair knockback and some spin, damage taken, easily pushed
--Diagnol Hit - Little knockback and alost 180 degree spin, damage taken, can't reallt be pushed but easily turned

Raptor Impact with Hellfire SPMA
-------UT3 Results--------
--Front Hit   - No knockback but entire SPMA jumped in the air, damage taken, cannot be pushed (how is it that it can be knocked into the air on impact but not pushed?)
--Side Hit    - No knockback and SPMA nearly flipped on it's side, damage taken, cannot be pushed
--Back Hit    - No knockback and SPMA treads are lifted off ground, damage taken, cannot be pushed
--Diagnol Hit - No knockback and Raptor is deflected, damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Medium knockback and lifted off front wheels, damage taken, can be pushed
--Side Hit    - Medium knockback and SPMA nearly flipped and Raptor went bottom up, damage taken, cannot be pushed but can be turned or flipped
--Back Hit    - Medium knockback and SPMA went bottom up, damage taken, can be pushed and made to go bottom up
--Diagnol Hit - Fair knockback and SPMA flipped, damage taken, can be pushed and turned and flipped
Extra Note: Neither SPMA seems to have taken damage but UT2004 Raptor took a lot while UT3 Raptor took little to none, this is another vehicle in UT2004 that the Raptor can nearly carry off

Raptor Impact with Paladin
-------UT3 Results--------
--Front Hit   - No or little knockback and Paladin is lifted off front wheels enough Raptor went under it, damage taken, cannot be pushed
--Side Hit    - No or little knockback and Paladin nearly flipped, damage taken, cannot be pushed
--Back Hit    - No or little knockback and Paldin was lifted off back wheels slightly, damage taken, cannot be pushed
--Diagnol Hit - No knockback and Paladin was slightly lifted off the ground and Raptor was deflected, damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Fair knockback and Paladin went bottom up, damage taken, can be pushed
--Side Hit    - Fair knockback and Paladin is fully rolled over (upside down and back to upright), damage taken, flips at slightest tap and can go on top of the Raptor and be carried off a bit before falling off
--Back Hit    - Fair knockback and Paladin is slightly lifted off back wheels, damage taken, slightest tap causes it to bottom up and go on top of the Raptor and be carrier off a little bit before it falls off
--Diagnol Hit - Fair knockback and Paladin went bottom up and completely turned around, damage taken, cannot be pushed without flipping it
Extra Note: In UT2004 both vehicles took a lot of damage, UT3 barely had any damage that I could see

Raptor Impact with Goliath
-------UT3 Results--------
--Front Hit   - No knockback and Raptor is deflected with loss of control, damage taken, can be very slowly pushed
--Side Hit    - No knockback but Goliath is lifted off the ground, damage taken, cannot be pushed
--Back Hit    - No knockback, damage taken, can be very slowly pushed
--Diagnol Hit - No knockback and Raptor is deflected, damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Little knockback and some spin, damage taken, can be slowly pushed
--Side Hit    - Little knockback and some spin, damage taken, can be slowly pushed and easily turned
--Back Hit    - Little knockback, damage taken, can be slowly pushed and easily turned
--Diagnol Hit - Little knockback and 90 degree spin, damage taken, can be slowly pushed and easily turned

Raptor Impact with Leviathan
-------UT3 Results--------
--Front Hit   - No knockback and on 1st time only slightly lifted off front wheels, damage taken, cannot be pushed
--Side Hit    - No knockback but almost flipped over, damage taken, cannot be pushed
--Back Hit    - No knockback, damage taken, cannot be pushed
--Diagnol Hit - No knockback and Raptor went vertical, damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - Fair knockback, damage taken, can be slowly pushed
--Side Hit    - Fair knockback and almost flips, damage taken, cannot be pushed
--Back Hit    - Fair knockback, damage taken, can be slowly pushed
--Diagnol Hit - Little knockback and front lifted into the air and turned 90 degrees, damage taken, can be slowly pushed
Extra Note: UT2004 Leviathan and Raptor were damaged, front wheel plates messed up on impact and wheels do not turn while being pushed (I'm just now noticing this so it might be true for previous vehicles as well)

Raptor Impact with Viper
-------UT3 Results--------
--Front Hit   - Medium knockback and is completely thrown into the air and turns almost 90 degrees, no damage taken, easily pushed
--Side Hit    - Medium knockback and is completely thrown into the air and should have landed on it's side, no damage taken, easily pushed
--Back Hit    - Medium knockback and is completely thrown into the air and turns 180 degrees, no damage taken, easily pushed
--Diagnol Hit - Medium knockback and is completely thrown into the air in front of Raptor and hit again, no damage taken, easily pushed
------UT2004 Results------ 
--Front Hit   - pending
--Side Hit    - pending
--Back Hit    - pending
--Diagnol Hit - pending

Raptor Impact with Scavenger
-------UT3 Results--------
--Front Hit   - Can't find a map that has both
--Side Hit    - 
--Back Hit    - 
--Diagnol Hit - 
------UT2004 Results------ 
--Front Hit   - pending
--Side Hit    - pending
--Back Hit    - pending
--Diagnol Hit - pending

Raptor Impact with Nightshade
-------UT3 Results--------
--Front Hit   - Medium knockback and completely tossed into the air, no damage taken, easily pushed
--Side Hit    - Fair knockback, no damage taken, easily pushed
--Back Hit    - Medium knockback and completely tossed into the air, no damage taken, easily pushed
--Diagnol Hit - Fair knockback, no damage taken, easily pushed
------UT2004 Results------ 
--Front Hit   - pending
--Side Hit    - pending
--Back Hit    - pending
--Diagnol Hit - pending

Raptor Impact with Fury
-------UT3 Results--------
--Front Hit   - Fair knockback, no damage taken, easily pushed
--Side Hit    - Fair knockback and turns 90 degrees
--Back Hit    - Fair knockback, no damage taken, easily pushed
--Diagnol Hit - Fair knockback to the side away from Raptor and slight turn, no damage taken, easily turned but difficult to push
------UT2004 Results------ 
--Front Hit   - pending
--Side Hit    - pending
--Back Hit    - pending
--Diagnol Hit - pending

Raptor Impact with Nemesis
-------UT3 Results--------
--Front Hit   - No knockback and front slightly lifted off ground, no damage taken, cannot be pushed
--Side Hit    - No knockback and slightly lifted off ground, no damage taken, cannot be pushed
--Back Hit    - No knockback and back very slightly lifted off ground, no damage taken, cannot be pushed
--Diagnol Hit - No knockback and Raptor deflected, no damage taken, cannot be pushed
------UT2004 Results------ 
--Front Hit   - pending
--Side Hit    - pending
--Back Hit    - pending
--Diagnol Hit - pending

Raptor Impact with Darkwalker
-------UT3 Results--------
--Front Hit   - Fair knockback and legs move with it, surprisingly no damage indicators, cannot be pushed but can be turned
--Side Hit    - No knockback and legs move, no damage taken, cannot be pushed but can be turned
--Back Hit    - No knockback and Raptor is deflected, no damage taken, cannot be pushed but can be turned
--Diagnol Hit - No knockback but Darkwalker turned and Raptor was deflected, no damage taken, cannot be pushed but can be turned
------UT2004 Results------ 
--Front Hit   - pending
--Side Hit    - pending
--Back Hit    - pending
--Diagnol Hit - pending

**Comment by HellDragon-HK** [comment]
Landing gear isn't working UT2004 and is making it hard to determine size but I think the Raptor might be already the right size maybe.  Any idea on why the landing gear doesn't work?

I tried to look at the animations but there aren't any in the editor so how does that work?

**Comment by 100GPing100** [comment]
@HellDragon-HK UE3 has a more advanced animation system, where you can have animations made based on variables, such as speed. That animation is one of them. The same thing happens for the manta's ailerons, in UT2004 I replicated this behaviour in code. In fact I remember doing some of this on the raptor too.

**Comment by HellDragon-HK** [comment]
The tail wings I'm guessing had to be done in code, if so you did a great job on those I think.

**Comment by 100GPing100** [comment]
@HellDragon-HK Yes that was it, the tail wings. If I remember correctly I still had to do the wings moving based on speed (stopped they point down, full speed they point backwards)

**Comment by HellDragon-HK** [comment]
Sorry GE but seems like a good spot to mention this....The Cicada has a similar need, the engine section is supposed to open and close based on speed but it currently just stays open all the time,

**Comment by 100GPing100** [comment]
@HellDragon-HK Yes that's something that's probably possible to do with code.

**Comment by HellDragon-HK** [comment]
@100GPing100 The Raptor in UT3 doesn't move it's main side wings until it's a ways off the ground, in UT2004 currently as soon as someone is in it they point downward and clip, is it possible to add a height requirement like UT3 has to fix the clipping?

**Comment by 100GPing100** [comment]
@HellDragon-HK Yes it is possible to make it only rotate the wings once it gets to a certain distance from the ground. It is also possible to make it fall slower when left in mid-air, it would be something like the Viper when ready to eject.

**Comment by GreatEmerald** [comment]
Yes, both wouldn't be too difficult to code. Which makes me wonder if in UT3 the Raptor always levitates, or only after exiting, i.e. if you push one off a cliff does it land slowly too, or if it falls...

**Comment by HellDragon-HK** [comment]
I thnk it slowly falls, I think I did that with the Cicada, hold on and I'll check as I'm in UT3 right now

**Comment by GreatEmerald** [comment]
OK, so it's triggered after an exit event, good to know.

**Comment by HellDragon-HK** [comment]
You guys disabled the jet stream visuals on purpose, why?  They are seen in UT3 (PC only though)

**Comment by HellDragon-HK** [comment]
@100GPing100 
Ugh well I tried adding GunsRotation.Yaw = -DriverViewYaw; to the funation Guns() and it does make them turn left and right but it's odd and too much rotation, any help?  

Same with Manta, adding your function Guns() allows the barrels to move up and down but trying to get left and right has same issue

**Comment by Unre-Alex** [comment]
I got always 40 to all vehicles.

**Comment by HellDragon-HK** [comment]
@Unre-Alex Ah perfect timing, I was about to ask you about this
	MaxThrustForce=70.0
	MaxRiseForce=70.0
	UpDamping=0.16
	MaxStrafeForce=40.0
	LatDamping=0.14

40 in UT2004 or UT3? I fired only 1 shot at every vehicle on Suspense (except the Raptor and I don't know why) and got mixed results

**Comment by Unre-Alex** [comment]
In UT3, make sure you're shooting precisely at the body, otherwise the values might be lower, and I hope you're not testing on ps3.

**Comment by HellDragon-HK** [comment]
I can't see the health numbers on PS3 so PC only and I'll try again

**Comment by HellDragon-HK** [comment]
Updated the Raptor collision to include most of the wings, I haven't had much time to test it yet so let me know if anything is off with it please

https://drive.google.com/open?id=1GAGc8gKLVdO66snC_5j2LVyGBn9QxSTj

**Comment by GreatEmerald** [comment]
I think the trails were removed because they did not correspond to the new model any more. Also there used to be four trails IIRC, and in UT3 there are only two. So it's good that you found the right numbers for them.

**Comment by HellDragon-HK** [comment]
According to my checkbox UT3 has 4 streamers which are all removed and trails are the engine flames which we have 2, UT2004 has 4 engine flames, I don't remember that but ok

**Comment by HellDragon-HK** [comment]
Raptor weapon fire un-fixable, I've tried WeaponOffset, WeaponFireOffset, WeaponFireLocation and all of them compile but are ignored in-game, specifically it's the Z value that is being ignored which is what is needed

**Comment by HellDragon-HK** [comment]
Added link for UT3 Drawscale ukx edit.....this one might have issues as the Raptor is now almost the size of a Goliath which is what it's supposed to be but as we all know the stock Raptor is tiny so I think we can expect some kind of spawn issues or something

**Comment by GreatEmerald** [comment]
The weapon fire location is controlled in code, see: https://github.com/GreatEmerald/UT3Vehicles/blob/master/Classes/UT3RaptorWeapon.uc#L205

**Comment by HellDragon-HK** [comment]
Ah you mean this?
CurrentFireOffset = (WeaponFireOffset * vect(1,0,0)) + (DualFireHack * vect(0,1,0));
Is the (1 the X  the next 0 the Y and the last 0) the Z?

**Comment by HellDragon-HK** [comment]
Movement vid added to bottom of task list
